src/engine/threaded_engine.cc

```diff
@@ -428,6 +447,14 @@ inline void ThreadedEngine::OnComplete(ThreadedOpr* threaded_opr) {
   for (auto&& i : threaded_opr->mutable_vars) {
     if (threaded_opr->opr_exception && *threaded_opr->opr_exception) {
       i->var_exception = threaded_opr->opr_exception;
+      // add current operator exceptions to global exceptions if not already
+      // added
+      auto it = std::find(global_exception_refs_.begin(),
```

Comment:
L452-L457 are used in three places. Can we make it a function?

Reply:
Thanks for the review ! I have moved it into a function.